uniflow-android-test/src/main/java/io/uniflow/android/test/SimpleTestObservers.kt

```diff
@@ -15,14 +15,23 @@ class TestObserver<T> : Observer<T> {
 }
 
 data class TestViewObserver(private val states: TestObserver<UIState>, private val events: TestObserver<Event<UIEvent>>) {
-    fun hasState(state: UIState): Boolean = states.elements.lastOrNull() == state
-    fun hasState(state: UIState, index: Int): Boolean = states.elements[index] == state
-    fun lastState(): UIState? = states.elements[states.elements.lastIndex]
-    fun statesCount() = states.elements.count()
-    fun hasEvent(event: UIEvent) = events.elements.lastOrNull() == event
-    fun hasEvent(event: UIEvent, index: Int) = events.elements[index] == event
-    fun eventsCount() = events.elements.count()
-    fun lastEvent(): UIEvent? = events.elements[events.elements.lastIndex].take()
+    fun hasState(state: UIState) = hasState(state, states.elements.lastIndex)
+    fun hasState(state: UIState, index: Int) = assert(states.elements[index] == state) { "should have state '$state' at $index but was '${states.elements[index]}'" }
+    fun hasStates(vararg states: UIState) = assert(this.states.elements == states) { "should have states [$states] but was [${this.states.elements}]" }
+    fun hasNoState() = assert(this.states.elements.isEmpty()) { "should have no state but was [${this.states.elements}]" }
+    val lastState: UIState?
+        get() = states.elements[states.elements.lastIndex]
+    val statesCount
+        get() = states.elements.count()
+
+    fun hasEvent(event: UIEvent) = hasEvent(event, events.elements.lastIndex)
```

Comment:
Is there a use case for checking `UIData` instead of `UIState` and `UIEvent`? As a user I might want to test that certain events and states are observed in a specific order, for example:

1. State A
1. Event 1
1. State B
1. Event 2
1. Event 3

This might be for a different issue or future PR, but does UniFlow support such a use case?

Reply:
you're clearly right, I need to review my copy :)